DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Menu.Script.txt

```diff
@@ -0,0 +1,105 @@
+	<label size="49 7" text="Quit" class="uiButtonElement" halign="center" valign="center2" focusareacolor1="0000" focusareacolor2="3ada" scriptevents="1" translate="0" textsize="1.2" data-action="Quit" />
+	</frame>
+	<!-- <frame pos="0 -35" class="uiContainer uiButton">
+	<quad size="30 5" style="Bgs1InRace" class="" substyle="BgColorContour" halign="center" valign="center2" opacity="0.5"/>
+	<label size="29 5" text="Classic Menu" class="uiButtonElement" halign="center" valign="center2" focusareacolor1="0000" focusareacolor2="3ada" scriptevents="1" translate="0" textsize="0.5" data-action="ClassicMenu" opacity="0.5" />
+	</frame> -->
+	</frame>
+	<script><!--
+
+	Void ShowMenu(Text MenuName) {
+		foreach (Elem in Page.MainFrame.Controls) {
+			Elem.Hide();
+			if (Elem.ControlId == MenuName) {
+				Elem.Show();
+				declare CMlFrame Frame = (Elem as CMlFrame);
+				foreach (i => Elem2 in Frame.Controls) {
+					Elem2.RelativeScale = 0.;
+					AnimMgr.Flush(Elem2);
+					AnimMgr.Add(Elem2, "<elem scale=\"1\" />", Now - 200 + 75*i, 500, CAnimManager::EAnimManagerEasing::ElasticOut);
+				}
+			}
+		}
+	}
+
+	main() {
+		declare Boolean MenuStatusOld = Playground.IsInGameMenuDisplayed;
+		while (True) {
+			yield;
+
+			if (Playground.IsInGameMenuDisplayed != MenuStatusOld ) {
+				MenuStatusOld = Playground.IsInGameMenuDisplayed;
+				if (Playground.IsInGameMenuDisplayed) {
+					ShowMenu("main-menu");
+				}
+			}
+			foreach(Event in PendingEvents) {
+				if (Event.Type == CMlScriptEvent::Type::KeyPress) {
+					if (Page.MainFrame.Visible == True && Event.KeyName == "Escape") {
+						CloseInGameMenu(CMlScriptIngame::EInGameMenuResult::Resume);
+					}
+				}
+				if (Event.Type == CMlScriptEvent::Type::MouseClick) {
+					declare Text action = Event.Control.DataAttributeGet("action");
+					switch (action) {
+						case "ShowMenu": ShowMenu("main-menu");
+						case "Setup": {
+							TriggerPageAction("maniaplanet:editsettings");
+						}
+						case "ApplySettings": {
+							declare CMlEntry name1 <=> (Page.GetFirstChild("team1name") as CMlEntry);
+							declare CMlEntry name2 <=> (Page.GetFirstChild("team2name") as CMlEntry);
+							declare value = "<script_settings><setting name=\"S_Team1Name\" type=\"text\" value=\""^ name1.Value ^"\"/><setting name=\"S_Team2Name\" type=\"text\" value=\""^ name2.Value ^"\"/></script_settings>";
+							Playground.RequestChangeModeScriptSettings(value);
+							CloseInGameMenu(CMlScriptIngame::EInGameMenuResult::Resume);
+						}
+						case "Resume": CloseInGameMenu(CMlScriptIngame::EInGameMenuResult::Resume);
+						case "Spectate": Playground.RequestSpectatorClient( !InputPlayer.RequestsSpectate );
+						case "ChangeTeam": {
+							declare Team = InputPlayer.RequestedClan;
+							if (Team == 1) { Playground.JoinTeam2(); }
+							else { Playground.JoinTeam1(); }
+							CloseInGameMenu(CMlScriptIngame::EInGameMenuResult::Resume);
+						}
+						case "Quit": Playground.QuitServer(True);
```

Comment:
What is the difference between Silent and Non-Silent?

Reply:
Well this is actually stupid, since now it's supposed to be silent, and it asks confirmation for exit.
I think it should ask, so i keep it in place for now.